Central-Server/API/jobs.py

```diff
@@ -0,0 +1,99 @@
+from flask import Blueprint, render_template, abort,jsonify, request, Response
+import database
+import auth
+import os.path
+from enum import Enum
+import sanitizers
+
+class JobStatus(Enum):
+    QUEUED = 0 # Job is queued and will be run on the next available TARS
+    RUNNING = 1 # Job is currently running on TARS
+    CANCELED = 2 # Job was canceled by someone #cancelculture
+    SUCCESS = 3 # Job was successfully run
+    FAILED_CRASHED = 4 # Job crashed on TARS
+    FAILED_COMPILE_ERROR = 5 # Job failed to compile
+    FAILED_TIMEOUT = 6 # Job timed out, cancelled #cancelculturestrikesagain
+    FAILED_OTHER = 7 # Job failed for some other reason
+    SETUP = 8
+
+def convert_job_info(job):
+    if job[8] > 7 or job[8] < 0:
+        job[8] = 7
+
+    return {"id": job[0],
+          "username": job[1],
+          "branch": job[2],
+          "hash": job[3],
+          "date_queue": job[5],
+          "date_start": job[6],
+          "date_end": job[7],
+          "status": JobStatus(job[8]).name,
+          }
+
+jobs_blueprint = Blueprint('jobs', __name__)
+@jobs_blueprint.route('/jobs/list')
+def list_jobs():
+    if not (auth.authenticate_request(request)):
+        abort(403)
+    # List out all the jobs in the database
+    conn = database.connect()
+    cursor = conn.cursor()
+    cursor.execute("SELECT * FROM hilsim_runs")
+    # Sort through the json and set status
+    job_list = []
+    for job in cursor.fetchall():
+        job_list.append(convert_job_info(job))
+    return jsonify(job_list)
+
+@jobs_blueprint.route('/jobs/<int:job_id>')
+def job_information(job_id):
+    # Get the jobs data from the job id
+    if (auth.authenticate_request(request) == False):

```

Comment:
nit: i think i prefer the `if not auth.authenticate_request(request)`

Reply:
I personally like this since it's more verbose, I know that the python standard is `not`, so I'm down to discuss this (applies to the below as well)